src/net/tls.cc

```diff
@@ -1479,6 +1479,18 @@ class session : public enable_lw_shared_from_this<session> {
                return put(std::move(p));
             });
         }
+
+        // We want to make sure that we call gnutls_record_send with as large
+        // packets as possible. This is because each call to gnutls_record_send
+        // translates to a sendmsg syscall. Further it results in larger TLS
+        // records which makes encryption/decryption faster. Hence to avoid
+        // cases where we would do an extra syscall for something like a 100
+        // bytes header we linearize the packet if it's below the max TLS record
+        // size.
+        if (p.nr_frags() > 1 && p.len() < 16384) {
```

Comment:
should this be <= ?

Reply:
`len()` is the total size of all the fragments, right?

Reply:
> should this be <= ?

Yes strictly speaking. Will change.

> len() is the total size of all the fragments, right?

Correct yes.